js/render/painter.js

```diff
@@ -105,6 +106,15 @@ Painter.prototype.setup = function() {
 };
 
 /*
+ * Set the lighting properties (used for extruded geometries).
+ */
+Painter.prototype.setLight = function(lightOptions) {
+    for (var key in lightOptions) {
+        this.light[key] = lightOptions[key];
+    }
+};
```

Comment:
I see `Painter` as a collection of GL utilities with no knowledge of map primitives and `Style` as the owner of state about the map style. Why did you choose to put this `light` state on `Painter` rather than `Style`?